packages/undo-manager/src/types.ts

```diff
@@ -0,0 +1,19 @@
+export type HistoryChange = {
+	from: any;
+	to: any;
+};
+
+export type HistoryChanges = {
+	id: string | Record< string, any >;
+	changes: Record< string, HistoryChange >;
+};
+
+export type HistoryRecord = Array< HistoryChanges >;
+
+export type UndoManager = {
+	addRecord: ( record: HistoryRecord, isCached: boolean ) => void;
```

Comment:
Why am I seeing `addRecord` being used with `HistoryChanges` everywhere, but here it only allows `HistoryRecord`? Sorry if I'm misreading this.

Reply:
I don't understand the question? `HistoryRecord` is an array of `HistoryChanges`. The former is a multi-object diff, the later is an object containing a diff for a single object.

Reply:
Yep, sorry I did misread it. Mixed if HistoryChange and HistoryRecord. It's a bit weird to me to have a singular word that is an array.